vllm/core/block/prefix_caching_block.py

```diff
@@ -191,11 +194,11 @@ def _free_block_id_for_block(self, block_id: BlockId,
 
         refcount = self._refcounter.decr(block_id)
 
-        # If no longer used, add the block to the unused cached blocks.
+        # If no longer used, add the block to the evictor.
         if refcount == 0:
-            assert block.content_hash not in self._unused_cached_blocks
             assert block.content_hash in self._cached_blocks
-            self._unused_cached_blocks[block.content_hash] = block_id
+            self.evictor.add(block.content_hash, block.block_id,
+                             block.num_tokens_total, block.last_accessed)
```

Comment:
What happens if a block is not computed yet?

Reply:
there is "assert block.content_hash in self._cached_blocks" to make sure this block shall have content_hash already. So we may safely claim it is a computed block?